src/backend/api/meals.js

```diff
@@ -2,14 +2,213 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// /api/meals	GET	Returns all meals
+
 router.get("/", async (request, response) => {
+  let query = knex("meal");
+
+  if (
+    "maxPrice" in request.query &&
+    typeof request.query.maxPrice === "string"
+  ) {
+    query = query.whereRaw("price <= ?", [request.query.maxPrice]);
+  }
+
+  if (request.query.availableReservations) {
+    query = query.join("reservation", "meal.id", "=", "reservation.meal_id");
+  }
+
+  if ("title" in request.query && typeof request.query.title === "string") {
+    query = query.whereRaw("title LIKE ?", [`%${request.query.title}%`]);
+  }
+
+  if (
+    "dateAfter" in request.query &&
+    typeof request.query.dateAfter === "string"
+  ) {
+    query = query.whereRaw("`when` > ?", [request.query.dateAfter]);
```

Comment:
I think it's a bit simpler to do something like this:

```suggestion
    query = query.where("when", ">", request.query.dateAfter);
```

Reply:
done